konnectivity-client/pkg/client/client_test.go

```diff
@@ -59,6 +59,49 @@ func TestDial(t *testing.T) {
 	}
 }
 
+func TestDialRace(t *testing.T) {
```

Comment:
We need to explain what is the race you are testing here. Now we know, but the code will be fixed and it will be hard to guess. It will be very hard to maintain this test otherwise.